### PR TITLE
Config dialog improvements

### LINKED ISSUE #472: Web UI doesn't save empty string values for settings
## Bug report

**Problem**

The web UI doesn't save empty string values for string settings. More or less coincidentally this made sense for all available settings until now, but with the upcoming merge of #469 a string setting (hostname) will be introduced that can be empty.

**Steps**

1. Load the web UI on a device with firmware that includes the hostname device setting. 
2. Open the device settings dialog.
3. Enter and apply a hostname if one is not already set. Then reopen the device settings dialog.
4. Remove the hostname and apply again.
5. Reopen the device settings dialog, and observe that the hostname that was configured before step 3 is still set.

**Notes**

To support the web UI knowing what settings are optional (i.e. may be empty), an optional "optional" property has been added to the output of the /settings/specs endpoint. That will also be merged with #469.

### LINKED ISSUE #476: Web UI: implement minimumValue and maximumValue boundary checking
The web UI currently does not apply `minimumValue` and/or `maximumValue` properties that may occur in the specification JSON objects for certain (numeric) settings. 

Ideally, the web UI should provide the user visual feedback when a value they enter is out of bounds from the perspective of `minimumValue` or `maximumValue`. Values that are out of bounds can also be excluded from POSTs to the /settings endpoints, as they will be ignored by the back-end anyway.

### PR DESCRIPTION
**Pull request by KeiranHines**
## Description
Fixed #472 Added support for `emptyAllowed` configurations to be saved.
Fixed #476 Added support for configuration to limit between min and max values for numeric input types. If no min/max is set the dataTypes min/max is used as a proxy.  

## Contributing requirements
* [x] I read the contribution guidelines in [CONTRIBUTING.md](http://github.com/PlummersSoftwareLLC/NightDriverStrip/blob/main/CONTRIBUTING.md).
* [x] I understand the BlinkenPerBit metric, and maximized it in this PR.
* [x] I selected `main` as the target branch.
* [x] All code herein is subjected to the license terms in [COPYING.txt](http://github.com/PlummersSoftwareLLC/NightDriverStrip/blob/main/COPYING.txt).

### CODE REVIEW

File: site/src/components/home/config/configDialog.jsx [hunk 1]
```diff
@@ -49,59 +49,61 @@ const ConfigInput = ({setting, updateData, updateError}) => {
...
             }
         }
     }, [value, updateData, setting]);
     const readOnly = !!setting.readOnly;
+    let min;
+    let max;
     switch(setting.type) {
     case settingType.Integer:
+        min = setting.minimumValue ? setting.minimumValue : (-2)**31;
+        max = setting.maximumValue ? setting.maximumValue : 2**31 -1;
```
Comment (rbergen) [inline site/src/components/home/config/configDialog.jsx:73]: Won't `setting.minimumValue` or `setting.maximumValue` also evaluate to `false` if they are 0 (which can very well be the case)? 

Same question obviously also applies to the `PositiveBigInteger` and `Float` cases.

Comment (rbergen) [reply site/src/components/home/config/configDialog.jsx:73]: @KeiranHines I had to look it up (my JS is a bit rusty), but 0 is indeed falsy. Which makes that I think we should apply an explicit strict inequality with `undefined` in these conditional expressions.

Comment (KeiranHines) [reply site/src/components/home/config/configDialog.jsx]: You are correct. That was a mistake. I have corrected it now.

**Comment by rbergen** [comment]
I have tested both the range checking and the empty string value submittal, and both seem to work well. 

@KeiranHines Combining the two, I was now thinking we should maybe also add an error condition when `emptyAllowed` is explicitly defined as `false` for a String setting, and its value is empty? Then we can choose to:

- Apply an empty String value (`emptyAllowed === true`)
- Reject it as an error (`emptyAllowed === false`), or 
- Silently ignore it, as we currently do for all but one (`emptyAllowed === undefined`). 

I'd have to make some changes in the backend to distinguish between "not set" and `false`, but I can do that independently of the implementation of this feature in the frontend. 

What are your thoughts?

**Comment by rbergen** [comment]
@KeiranHines I was wondering if you could (just) comment on the idea I posted concerning empty strings? It's no problem at all if you can't (yet) work on the front-end implementation of it, but if you agree it's worth implementing at some point then I can already take care of the back-end part of it. I can do that because the current front-end implementation is compatible with both the current setup and the one I'm proposing.

**Comment by KeiranHines** [comment]
I could see a use case for things like YouTube channel in the subscriber count effect. If you add the backend it's a single if statement to add it into the front-end and I'll add it when available.

**Comment by rbergen** [comment]
@KeiranHines I've just merged #490, so you can add the check for `emptyAllowed === false` whenever you're ready.

**Pull request merged**